### DISCUSSION TITLE
ValidatedNotNull and why not static?

### DISCUSSION THREAD

**Discussion by thewayibyte** [ardalis/GuardClauses#122] [Q&A] [answered]
Hello,

two questions from my side: 
1. Why should one use this ValidatedNotNull attribute? What it does? Within the examples I do not see usage of it.
2. Why the whole Guard class is not just a static class with static methods within? The creation of this class looks like normal class and I do not see any reason for such behaviour. 

Thanks kindly :)

**Comment by egreiner** [comment]
regarding 2: **Extensibility**

This is a good example for the open-closed principle
Everybody can extend to their needs easily without touching the original framework

hth

**Reply by thewayibyte** [reply]
Yeah I get that, but I think that this is one of those cases where you have to decide, do I want to be "SOLID" or do I want to have "easy" functionality. 
Why I asked about making this as static class. Well, because when you have the current implementation, then I have to create this class's object each time or at least receive the class's instance, so this was my reason for this suggestion :)

I agree that a lot of people will have their own need to implement additional Guard methods and it is nice that they can derive from that implementation.
So maybe there is some other way, and my way of thinking will be changed to something better? :D

**Comment by ardalis** [comment, marked as answer]
The `Guard` class implements the `IGuardClause` interface; static classes cannot implement interfaces. So, that's the main reason. But notice that you don't instantiate `Guard`. Nowhere in the samples do you see `var guard = new Guard()` or anything like that. So is there an issue you're seeing because it's not a static class?
